pkg/cloud/common/interfaces.go

```diff
@@ -0,0 +1,25 @@
+package common
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CloudProviderAssets represent cloud provider specific assets object. Such as Azure/Aws/OpenStack/etc.
+// Intended to hide provider specific resource management, manifests checks and rendering behind this interface.
+type CloudProviderAssets interface {
+	// GetRenderedResources intended to return ready to apply list of resources.
+	// Expected that resources returned by this method is completely prepared and do not need additional handling.
+	GetRenderedResources() ([]client.Object, error)
+}
+
+// TemplateBasedAssets interface for specify assets object which leveraging golang standard template/text library
+// under the hood.
+type TemplateBasedAssets interface {
+	CloudProviderAssets
+	// GetTemplates returns list of templates for further rendering.
+	GetTemplates() []ObjectTemplate
+	// GetTemplateValues returns map with values for further templates rendering.
+	// This map is global for all templates within assets object and intended to be applied to each ObjectTemplate
+	// for single CloudProviderAssets
+	GetTemplateValues() (TemplateValues, error)
```

Comment:
Where are these used? Don't we just call `GetRenderedResources` and cut out the middle man?

Reply:
https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/110/files#diff-5e1ede80f1538a3bc199f37052f71e420c9aa8505982c96ad0119f829ab67da6R28

Idea behind this method - encapsulate knowledge about needed variables inside `Assets` object and have a proper place to construct such map. 

Hm, looking on your comment on `RenderTemplate` function makes me think that this interface might be rethinked a bit.